utxorpc/sync_test.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2025 Blink Labs Software
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utxorpc
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"testing"
+
+	"connectrpc.com/connect"
+	"github.com/blinklabs-io/dingo/chain"
+	"github.com/blinklabs-io/dingo/database"
+	"github.com/blinklabs-io/dingo/database/models"
+	"github.com/blinklabs-io/dingo/ledger"
+	ocommon "github.com/blinklabs-io/gouroboros/protocol/common"
+	sync "github.com/utxorpc/go-codegen/utxorpc/v1alpha/sync"
+)
+
+// setupTestSyncServer creates a test sync server with minimal configuration
+func setupTestSyncServer(t *testing.T) (*syncServiceServer, *database.Database, *chain.Chain) {
+	t.Helper()
+
+	// Create test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	primaryChain := cm.PrimaryChain()
+
+	// Create ledger state
+	ls := &ledger.LedgerState{}
+	// Use reflection-like approach by creating minimal struct
+	// In real code, this would use proper initialization
+	// For testing, we'll set the necessary fields directly
+	// Note: This is simplified for testing purposes
+
+	// Create utxorpc config
+	utxorpcCfg := UtxorpcConfig{
+		Logger: slog.New(slog.NewJSONHandler(io.Discard, nil)),
+	}
+
+	utxorpc := &Utxorpc{
+		config: utxorpcCfg,
+	}
+
+	server := &syncServiceServer{
+		utxorpc: utxorpc,
+	}
+
+	return server, db, primaryChain
+}
+
+// TestFetchBlock_WithValidBlockRef tests FetchBlock with valid block references
+func TestFetchBlock_WithValidBlockRef(t *testing.T) {
+	server, db, primaryChain := setupTestSyncServer(t)
+	defer db.Close()
+
+	// Create and add test block
+	testBlock := models.Block{
+		Hash:     []byte{0x01, 0x02, 0x03, 0x04},
+		Slot:     12345,
+		Number:   100,
+		PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+		Type:     5,
+	}
+
+	if err := primaryChain.AddBlock(testBlock, nil); err != nil {
```

Comment:
@coderabbitai This should be a "github.com/blinklabs-io/gouroboros/ledger/common".Block not a models.Block struct. You can use Decode() on models.Block to get a ledger.Block suitable for use here.